conda/recipes/libraft/build_libraft_tests.sh

```diff
@@ -1,5 +1,8 @@
 #!/usr/bin/env bash
 # Copyright (c) 2022-2023, NVIDIA CORPORATION.
 
+# Just building template so we verify it uses libraft.so and fail if it doesn't build
+./build.sh template
```

Comment:
Cool way to test this! It probably is insignificant increase in this test's time, right, since all we'll check if the template can link against `libraft.so`

Reply:
Correct. I verified (check my tasks above in the description) that it reuses the already compiled `libraft.so` instead of rebuilding everything.

Reply:
Building non-tests in the `libraft-tests` build script seems like a bit of an anti-pattern.

This new raft template project reminds me of the `libcudf-example` package that is built [here](https://github.com/rapidsai/cudf/blob/a0473cf5b7c52b157584bd02e5c60897c05a57f8/conda/recipes/libcudf/meta.yaml#L347-L373) by `cudf`.

For consistency with that example, can we move this to its own package within `libraft`'s `meta.yaml` file?

Reply:
Sure. Yeah actually my intention here wasn't even to package up the example at all, but rather just to build it somewhere in CI so that we know when it fails. This particular build is not meant to be an example so much as a copy/paste template for users to start their projects from (so it shouldn't have any deployed artifact at all). Does that change anything here? 

I like that libcudf examples too, and it's actually making me think we should be providing some raft examples as well, but as an actual package like libcudf.

Just to be clear- I don't mind separating out the template skeleton example, but I don't plan to deploy any of it.

Reply:
Gotcha. Thanks for the clarification.

I don't think it changes much.

You can prevent the new conda package from being uploaded to Anaconda.org by passing it's name to this `skip_upload_pkgs` key in the `upload-conda` job of `build.yaml` like `cudf` does: https://github.com/rapidsai/cudf/blob/a0473cf5b7c52b157584bd02e5c60897c05a57f8/.github/workflows/build.yaml#L55

Reply:
Okay sounds good. I'll separate this out. 

I think we might need to do the same for our new Ann benchmarks too, so I'll go ahead and add to that pr as well.